packages/thirdweb/src/wallets/injected/index.ts

```diff
@@ -200,10 +200,15 @@ async function onConnect(
   }
 
   function onAccountsChanged(accounts: string[]) {
-    if (accounts.length === 0) {
-      onDisconnect();
+    if (accounts[0]) {
+      const newAccount = {
+        ...account,
+        address: getAddress(accounts[0]),
```

Comment:
is there an assumption that the 'new' account will always be accounts[0] ?

also kind of weird to spread the old account functions and just override the address, but i guess in this case it'll work

Reply:
Yes - the first one is the "active" account if the wallet has multiple accounts ( in case of metamask ) 

Re: the spread: We can also simply mutate the `address` property - but if we do that -  the account object will not trigger a re-render in react - so we clone the object and change the address since that's the only change we need to make